src/DlibDotNet/ImageTransforms/ImageGradients.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using DlibDotNet.Extensions;
+
+// ReSharper disable once CheckNamespace
+namespace DlibDotNet
+{
+
+    public sealed class ImageGradients : DlibObject
+    {
+
+        #region Constructors
+
+        public ImageGradients(long scale = 1)
+        {
+            this.NativePtr = NativeMethods.get_image_gradients(scale);
+        }
+
+        #endregion
+        
+        #region Methods
+
+        public Rectangle GetGradientX(Array2DBase image, Array2D<float> gradient)
+        {
+            return InvokeGetGradient(image, gradient, NativeMethods.image_gradients_gradient_x);
+        }
+        public Rectangle GetGradientY(Array2DBase image, Array2D<float> gradient)
+        {
+            return InvokeGetGradient(image, gradient, NativeMethods.image_gradients_gradient_y);
+        }
+
+        public Rectangle GetGradientXY(Array2DBase image, Array2D<float> gradient)
+        {
+            return InvokeGetGradient(image, gradient, NativeMethods.image_gradients_gradient_xy);
+        }
+
+        public Rectangle GetGradientXX(Array2DBase image, Array2D<float> gradient)
+        {
+            return InvokeGetGradient(image, gradient, NativeMethods.image_gradients_gradient_xx);
+        }
+
+        public Rectangle GetGradientYY(Array2DBase image, Array2D<float> gradient)
+        {
+            return InvokeGetGradient(image, gradient, NativeMethods.image_gradients_gradient_yy);
+        }
+
+
+        private Rectangle InvokeGetGradient(Array2DBase image, Array2D<float> gradient, Func<IntPtr, NativeMethods.Array2DType, IntPtr, IntPtr, IntPtr, NativeMethods.ErrorType> gradientMethod)
+        {
+            this.ThrowIfDisposed();
+
+            if (image == null)
+                throw new ArgumentNullException(nameof(image));
+
+            if (gradient == null)
+                throw new ArgumentNullException(nameof(image));
```

Comment:
nameof(gradient)

Reply:
And 
````
image.ThrowIfDisposed();
graduent.ThrowIfDisposed();
````

Reply:
And, you created feature branch but it does not derive from develop branch.
Why don't you use gitflow?
By chance, does github PR force to create branch from master?

Reply:
Thanks for the feedback. Done. 

Regarding the branch, I don't understand? 
Should I branch from development instead of master?

Reply:
> Should I branch from development instead of master?
Yes. I adopt gitflow development model.
I'm very sorry for the inconvenience.